### PR TITLE
Upgrade lsp4j version to 0.7.1 and fixed errors it introduced

### LINKED ISSUE #370: Move to lsp4j 0.7.1
Move to `lsp4j 0.7.1` (last release) This change will provide compilation problem because some signature changes, but not a very hard task

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Addition to partial fix of https://github.com/angelozerr/lsp4xml/pull/372

Fixes #370

### CODE REVIEW

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java [hunk 1]
```diff
@@ -303,14 +304,14 @@ public static void assertDiagnostics(List<Diagnostic> actual, Diagnostic... expe
 	public static void assertDiagnostics(List<Diagnostic> actual, List<Diagnostic> expected, boolean filter) {
 		List<Diagnostic> received = actual;
 		final boolean filterMessage;
-		if(expected != null && !expected.isEmpty() && expected.get(0).getMessage() != null) {
+		if(expected != null && !expected.isEmpty() && expected.get(0).getMessage() != null && expected.get(0).getMessage().isEmpty() == false) {
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java]: better to use concise code like 

!expected.get(0).getMessage().isEmpty()

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/prolog/PrologModel.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ private static void createCompletionItem(String attrName, boolean canSupportSnip
 		MarkupContent markup = new MarkupContent();
 		markup.setKind(MarkupKind.MARKDOWN);
 		
-		markup.setValue(documentation);
+		markup.setValue(documentation != null ? documentation : "");
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/prolog/PrologModel.java]: deserves a StringUtils.defaultString(documentation)

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java (hunk 1 above)
Comment (fbricon) [reply org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java]: `if(expected != null && !expected.isEmpty() && !StringUtils.isEmpty(expected.get(0).getMessage()))`

**Comment by NikolasKomonen** [comment]
@fbricon updated

**Pull request merged**